### PR TITLE
Add option to only send output to view archive

### PR DESCRIPTION
**Pull request by akashdhruv**
@kweide Can you review this?

### CODE REVIEW

**Review by kweide — COMMENTED** [review]
I don't know whether or how this will work as intended - don't quite understand how communication of these flags to the underlying `flashTest.py` works. I assume you will test it...

Assuming it does, I find the naming of the flags (and variables) potentially confusing.

The name `--view-archive` could be misunderstood as an instruction to `flashxtest run-suite` to (somehow) "view the archive" - whatever that could mean.
Maybe `--save-viewarchive` is a better name.
Then for symmetry, there should also be `--save-mainarchive` to do what the name obviously suggests.
None of those two should have "only" in their description. `--save-viewarchive` should only effect the view archive, and `--save-mainarchive`  should only effect the main archive; but neither of them would effect the other kind of archive (as the word "only" would suggest).
Then `--archive` could be redefined as just a shortcut for `---save-mainarchive --save-viewarchive`.

**Comment by akashdhruv** [comment]
Done

**Review by kweide — COMMENTED** [review]
Is there a way to set these flags to `False`? If yess, what's the syntax for that?
E.g., `--archive=False`, or `-archive no`, or `--archive=0`, or `--no-archive`, etc...

I am trying to make sense of the language on overriding.
I would not say that `--archive` overrides `--save-viewarchive`, for example. (Nor vice versa.) They sort of confirm each other.

However, `--archive=no` (if it exists) would override `--save-viewarchive[=yes]`.

Or does the `click` package allow a choice for each flag whether it allows a negative form or not?  If yes, I would say that `--save-viewarchive` and `--save-mainarchive` should both allow `yes` and `no`, and `--archive` should always be positive and act as if it were a shortcut for `--save-viewarchive[=yes] --save-mainarchive[=yes]`, possible a shortcut that always has priority [independent of the order in which flags occur, if your like].

File: FlashXTest/cli/_cli.py [hunk 1]
```diff
@@ -177,17 +178,35 @@ def setup_suite(suitelist, overwrite, add_setup_opts, seed_from_info):
 
 
 @flashxtest.command(name="run-suite")
-@click.option("--archive", is_flag=True, help="Save results to main archive")
-def run_suite(archive):
+@click.option(
+    "--archive", is_flag=True, help="Save results to both main and view archive"
+)
+@click.option(
+    "--save-viewarchive", is_flag=True, help="Save results to view archive only"
```
Comment (kweide) [inline FlashXTest/cli/_cli.py:185]: should not have only in the help string (the way I think of it)

**Comment by akashdhruv** [comment]
> Is there a way to set these flags to `False`? If yes, what's the syntax for that? E.g., `--archive=False`, or `-archive no`, or `--archive=0`, or `--no-archive`, etc...
> 
> I am trying to make sense of the language on overriding. I would not say that `--archive` overrides `--save-viewarchive`, for example. (Nor vice versa.) They sort of confirm each other.
> 
> However, `--archive=no` (if it exists) would override `--save-viewarchive[=yes]`.
> 
> Or does the `click` package allow a choice for each flag whether it allows a negative form or not? If yes, I would say that `--save-viewarchive` and `--save-mainarchive` should both allow `yes` and `no`, and `--archive` should always be positive and act as if it were a shortcut for `--save-viewarchive[=yes] --save-mainarchive[=yes]`, possible a shortcut that always has priority [independent of the order in which flags occur, if your like].

Simplified the behavior

```
Usage: flashxtest run-suite [OPTIONS]

  Run the test suite using "test.info".

  This command runs all the tests defined in
  "test.info", and conveys errors.

Options:
  --archive           Archive test results  [default: False]
  --skip-viewarchive  Skip saving results to view archive  [default: False]
  --skip-mainarchive  Skip saving results to main archive  [default: False]
  --help              Show this message and exit.
```

**Comment by kweide** [comment]
Ok for the changed code.

I was hoping you would give me an answer to the question:
> Is there a way to set these flags to False? If yes, what's the syntax for that? E.g., --archive=False, or -archive no, or --archive=0, or --no-archive, etc...

**Comment by akashdhruv** [comment]
> I was hoping you would give me an answer to the question:
> 
> > Is there a way to set these flags to False? If yes, what's the syntax for that? E.g., --archive=False, or -archive no, or --archive=0, or --no-archive, etc...

I have been reading click documentation, and here is what I have found

--archive=False, not possible

--archive=no, possible as a text input, but we will have to add extra checks to ensure that input is either yes/no and not unrecognized value

--archive=0, same as --archive=no

There is way to set --archive/--no-archive (https://click.palletsprojects.com/en/8.1.x/options/) but I decided to go with a more straightforward logic.

**Comment by kweide** [comment]
Thanks!

**Pull request merged**